octodns/__init__.py

```diff
@@ -1,6 +1,25 @@
-'OctoDNS: DNS as code - Tools for managing DNS across multiple providers'
-
 from __future__ import absolute_import, division, print_function, \
     unicode_literals
+import pkg_resources
+from os import path
+from setuptools.config import read_configuration
+
+
+def _extract_version(package_name):
+    try:
+        return pkg_resources.get_distribution(package_name).version
+    except pkg_resources.DistributionNotFound:
+        _conf = read_configuration(
+            path.join(
+                path.dirname(path.dirname(__file__)),
+                'setup.cfg'
+            )
+        )
+        return _conf['metadata']['version']
+
+
+__version__ = _extract_version('octodns')
```

Comment:
I think it'd be nice to avoid loading the version since it now involves some "work" in cases where it's not going to be used. Something like the following would do that, but there may be slightly cleaner ways to pull it off.

```python
#!/usr/bin/env python

class _VersionLoader(object):
    version = None

    @property
    def __version__(self):
        if self.version == None:
            # do the work to load it
            self.version = 42
        return self.version


__VERSION__ = _VersionLoader().__version__

if __name__ == "__main__":
    print(__VERSION__)
```

```bash
otter:~ ross$ python /tmp/foo.py
42
```